core/core.go

```diff
@@ -307,12 +306,22 @@ func New(config Config) (*App, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// Remove any old mini headers that might be lingering in the database.
+	// HACK(albrow): This is a workaround for an issue where miniheaders are
+	// not removed from the database. This issue has been fixed and this hack
+	// should be removed after the next release.
+	err = meshDB.PruneMiniHeadersAboveRetentionLimit()
+	if err != nil {
+		return nil, err
+	}
+
 	topics := orderwatch.GetRelevantTopics()
 	miniHeaders, err := meshDB.FindAllMiniHeadersSortedByNumber()
 	if err != nil {
 		return nil, err
 	}
-	stack := simplestack.New(blockWatcherRetentionLimit, miniHeaders)
+	stack := simplestack.New(meshDB.MiniHeaderRetentionLimit, miniHeaders)
```

Comment:
I think it's cleaner if `MiniHeaderRetentionLimit` is declared in `core` and passed through to both `meshdb` and `simplestack`. The `core` package should be the glue that connects everything together.

Reply:
Decided to just take this on as technical debt for now. We should address it in the future, possibly at the same time as cleaning up some other parts of the MiniHeaders collection.